src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -48,39 +72,6 @@ class UrlGenerator implements UrlGeneratorInterface, ConfigurableRequirementsInt
     protected $logger;
 
     /**
-     * This array defines the characters (besides alphanumeric ones) that will not be percent-encoded in the path segment of the generated URL.
-     *
-     * PHP's rawurlencode() encodes all chars except "a-zA-Z0-9-._~" according to RFC 3986. But we want to allow some chars
-     * to be used in their literal form (reasons below). Other chars inside the path must of course be encoded, e.g.
-     * "?" and "#" (would be interpreted wrongly as query and fragment identifier),
-     * "'" and """ (are used as delimiters in HTML).
-     */
-    protected $decodedChars = array(
-        // the slash can be used to designate a hierarchical structure and we want allow using it with this meaning
-        // some webservers don't allow the slash in encoded form in the path for security reasons anyway
-        // see http://stackoverflow.com/questions/4069002/http-400-if-2f-part-of-get-url-in-jboss
```

Comment:
I agree that this is not the best way of doing so, but it's still a way that people used to extend upon: http://stackoverflow.com/questions/24085126/patching-extending-route-urlgenerator-for-url-rewriting-to-replace-all-spaces-wi

It shouldn't be hard to create a BC layer though, `if (!empty($this->decodedChars)) { /* @trigger_error(...)`.